File: android/interactable/src/main/java/com/wix/interactable/InteractableArea.java
package com.wix.interactable;

public class InteractableArea {
    private float top;
    private float left;
    private float bottom;
    private float right;
    private float bounce;
    private boolean haptics;

    public InteractableArea(float top, float left, float bottom, float right, float bounce, boolean haptics) {
        this.top = top;
        this.left = left;
        this.bottom = bottom;
        this.right = right;
        this.bounce = bounce;
        this.haptics = haptics;
    }


    public float getTop() {
        return top;
    }

    public float getLeft() {
        return left;
    }

    public float getBottom() {
        return bottom;
    }

    public float getRight() {
        return right;
    }

    public float getBounce() {
        return bounce;
    }

    public boolean isHaptic() {
        return haptics;
    }
}

File: ios/Interactable/InteractableArea.m
//
//  InteractableArea.m
//  Interactable
//
//  Created by Tal Kol on 2/6/17.
//  Copyright © 2017 Wix. All rights reserved.
//

#import "InteractableArea.h"

@implementation InteractableArea

- (id)copyWithZone:(__unused NSZone *)zone
{
    return self;
}

- (BOOL)pointInside:(CGPoint)point withOrigin:(CGPoint)origin
{
    CGFloat cx = point.x - origin.x;
    CGFloat cy = point.y - origin.y;
    
    if (cx < self.left) return NO;
    if (cx > self.right) return NO;
    
    if (cy < self.top) return NO;
    if (cy > self.bottom) return NO;
    
    return YES;
}

@end
